### ISSUE TITLE
[prefer-mock-promise-shorthand] incorrectly reports on `mockImplementation`s with parameters

### ISSUE THREAD

**Issue by wheeler** [jest-community/eslint-plugin-jest#1184]
The new rule `prefer-mock-promise-shorthand` is cool! In trying to use the rule/autofix it looks like it has a bug around subtleties of `mockImplementation` parameters.

Here is what the new rule is currently identifying and "fixing"
```diff
- jest.spyOn(Thingy, 'method').mockImplementation(importantParameter => Promise.resolve(importantParameter));
+ jest.spyOn(Thingy, 'method').mockResolvedValue(importantParameter); 
```
this leads to its a separate basic eslint error:
>error  'importantParameter' is not defined  `no-undef`

I think that instances of `mockImplementation` with parameters should probably be ignored by this rule...

**Comment by wheeler** [comment]
cc @G-Rath @SimenB bug in new [prefer-mock-promise-shorthand] rule

**Comment by Retracted81** [comment]
Yeah where at I've sent bug report and the file how do I get it controlled
